### ISSUE TITLE
`@rx-angular/cdk/zone-less/rxjs` doesn't play well with RxJS TestScheduler

### ISSUE THREAD

**Issue by exequiel09** [rx-angular/rx-angular#1372]
# Description

I'm using the `@rx-angular/cdk/zone-less/rxjs` package. Specifically the `timer` observable creational function. When I was using version `1.0.0-beta.2` it wasn't broken. My unit tests started to act weird when I upgraded to version `1.0.0-beta.3`. I think this is related to the refactoring made to schedulers inside the CDK package.

# Steps to Reproduce the Issue

Reproduction repo: https://github.com/exequiel09/rxa-cdk-scheduler-issue/tree/ea21ee5079879eb25f278d987ce2e4cf54a2c0c5/apps/rxa-cdk-scheduler-issue/src/app/backoff

Just run `npx nx run rxa-cdk-scheduler-issue:test`

The example in the repo uses a copy of [`backoff-rxjs`](https://www.npmjs.com/package/backoff-rxjs) that's patched with `@rx-angular/cdk` to use the `timer` observable.

# Environment

Angular: 14
RxJS: 7.4

Also, the actual project uses RxJS 6 and Angular 13 and it happens. The repro above is just a minimal reproduction of the issue.

![Screen Shot 2022-06-14 at 5 15 44 PM](https://user-images.githubusercontent.com/1289472/173541729-b5f7d2f8-602d-424b-9d9a-e0f6f4aeb3f8.png)

**Comment by exequiel09** [comment]
Getting 404 @edbzn 

![Screen Shot 2022-06-14 at 5 27 05 PM](https://user-images.githubusercontent.com/1289472/173544224-fd5f163d-207e-4528-926a-c8de309c2683.png)

**Comment by exequiel09** [comment]
Should be: https://www.npmjs.com/package/rxjs-zone-less

**Comment by edbzn** [comment]
@exequiel09 Did you try using the `RxTestScheduler` from `rxjs-zone-less`?

```ts
import { catchError, EMPTY, throwError } from 'rxjs';
import { RxTestScheduler } from 'rxjs-zone-less';
import { retryBackoff } from './backoff-zoneless';

describe('RxJS Timer (via rxjs-zone-less)', () => {
  let testScheduler: RxTestScheduler;

  beforeEach(() => {
    testScheduler = new RxTestScheduler((actual, expected) => {
      expect(actual).toEqual(expected);
    });
  });

  it('should throw at 1400ms', () => {
    testScheduler.run((helpers: any) => {
      const { expectObservable } = helpers;
      const source$ = throwError(() => new Error('message')).pipe(
        retryBackoff({
          initialInterval: 200,
          maxRetries: 3,
          resetOnSuccess: true,
        }),

        catchError(() => EMPTY)
      );

      expectObservable(source$).toBe('1400ms |');
    });
  });
});
```

**Comment by exequiel09** [comment]
Not yet. will try.

**Comment by exequiel09** [comment]
@edbzn it passed. not sure what the `RxTestScheduler` does tho. I may need to read the source to learn why we're shipping custom test scheduler

![Screen Shot 2022-06-14 at 6 28 51 PM](https://user-images.githubusercontent.com/1289472/173556804-04e37cd7-75f8-4dac-8eaa-ea8473b55611.png)

**Comment by hoebbelsB** [comment]
@exequiel09 the `TestScheduler` internally replaces all methods timer events with custom functions so that you have control when e.g. the animationFrame fires. rxjs calls this `delegates`. Because the rx-angular schedulers are not able to use the same delegates, we had to provide our own + ship a custom testscheduler to control them.

The idea was to provide the same dx as rxjs does with the testscheduler. I think it will just be super hard to mix them ^^

Regarding the new package, yes it is as safe as it was before. It's the same source. We deprecated the `zone-less/rxjs` package in rx-angular because the maintenance is too much of an effort.

Is your issue resolved?
